PuzzleSlider/src/Engine/Engine.java

```diff
@@ -20,6 +20,7 @@ public Engine(int size) throws InvalidArgumentException {
 		mGameData = new GameData(size);
 		mGameData.initialize();
 		shuffleNewGame(1000);
+		mGameData.initVariables();
```

Comment:
Why are you initializing variables outside of class? Any time you are creating new instance of class you have to remember that. I hope there is better way to that in `initialize()` for example ... Or am I wrong?